code/modules/mob/living/carbon/xenomorph/castes/hivemind/hivemind.dm

```diff
@@ -19,22 +19,25 @@
 	tier = XENO_TIER_ZERO
 	upgrade = XENO_UPGRADE_ZERO
 
-	mouse_opacity = MOUSE_OPACITY_OPAQUE
+	mouse_opacity = MOUSE_OPACITY_TRANSPARENT
 	lighting_alpha = LIGHTING_PLANE_ALPHA_MOSTLY_INVISIBLE
 	see_invisible = SEE_INVISIBLE_LIVING
-	invisibility = INVISIBILITY_OBSERVER
+	invisibility = INVISIBILITY_ABSTRACT
 	sight = SEE_MOBS|SEE_TURFS|SEE_OBJS
 	see_in_dark = 8
 	move_on_shuttle = TRUE
 
 	hud_type = /datum/hud/hivemind
 	hud_possible = list(PLASMA_HUD, PHEROMONE_HUD, QUEEN_OVERWATCH_HUD)
 
-	var/obj/effect/alien/weeds/node/hivemindcore/core
+	var/obj/effect/alien/hivemindcore/core
 
-/mob/living/carbon/xenomorph/hivemind/Initialize(mapload, obj/effect/alien/weeds/node/hivemindcore/hmcore)
+/mob/living/carbon/xenomorph/hivemind/Initialize(mapload)
 	. = ..()
-	core = hmcore
+	var/mob/living/carbon/xenomorph/hivemind/newmind = src
+	var/obj/effect/alien/hivemindcore/newcore = new /obj/effect/alien/hivemindcore(loc)
+	newcore.parent = src
+	newmind.core = newcore
```

Comment:
```suggestion
	core = new(loc)
	core .parent = src
```

Reply:
we had something similar before and it broke when parented, making hiveminds immortal. The current system looks bad but works better.